query/security/wcpm/src/main/java/org/codice/ddf/admin/security/wcpm/actions/persist/SaveWhitelistContexts.java

```diff
@@ -56,22 +54,24 @@ public SaveWhitelistContexts(ConfiguratorFactory configuratorFactory) {
 
     @Override
     public ListField<ContextPath> performAction() {
+
         ListField<ContextPath> preUpdateWhitelistContexts = new ListFieldImpl<>(ContextPath.class);
-        for (String path :  getWhitelistContexts(configuratorFactory.getConfigReader())) {
+        for (String path : getWhitelistContexts(configuratorFactory.getConfigReader())) {
             preUpdateWhitelistContexts.add(new ContextPath(path));
         }
 
         Configurator configurator = configuratorFactory.getConfigurator();
-        configurator.updateConfigFile(POLICY_MANAGER_PID,
-                whiteListToPolicyManagerProps(contexts),
+        configurator.updateConfigFile(PolicyManagerServiceProperties.POLICY_MANAGER_PID,
+                new PolicyManagerServiceProperties().whiteListToPolicyManagerProps(contexts),
                 true);
 
         OperationReport configReport = configurator.commit(
                 "Whitelist Contexts saved with details: {}",
                 contexts.toString());
 
+        // TODO: tbatie - 4/23/17 - We need to discuss whether we want to return values even if it fails
```

Comment:
Yeah - we should do that. Just need to make sure nothing in the system is changed if errors happen.